R/differential_expression.R

```diff
@@ -294,31 +294,33 @@ plot_volcano <- function(
   
   checkmate::assertDataFrame(regression_significance)
   stopifnot("term" %in% colnames(regression_significance))
-
+  
   effect_var <- dplyr::case_when(
     "estimate" %in% colnames(regression_significance) ~ "estimate",
     "sumsq" %in% colnames(regression_significance) ~ "sumsq",
     TRUE ~ NA_character_
   )
-
+  
   if (is.na(effect_var)) {
     stop("volcano plot cannot be generated due to unknown test")
   }
-
+  
   regression_significance %>%
     dplyr::filter(!is.na(p.value)) %>%
     dplyr::mutate(
       p.value.trans = trans_pvalues(p.value, max_p_trans = max_p_trans),
       is_discovery = qvalue < FDR_cutoff
     ) %>%
     ggplot(aes_string(x = effect_var)) +
-    geom_point(aes(y = p.value.trans, color = is_discovery)) +
+    {if ("compoundName" %in% colnames(regression_significance))
```

Comment:
This is fine to me since it will allow for plotly compatibility but won't affect the static plot. If you wanted to add labels in for the static plot that would probably be useful and I would recommend passing names or groupIds to label in as an optional argument